packages/react/src/components/Tree/TreeItem.tsx

```diff
@@ -78,20 +90,24 @@ class TreeItem extends AutoControlledComponent<ReactProps<TreeItemProps>, TreeIt
     accessibility: defaultBehavior,
   }
 
+  public state = {
+    open: false,
+  }
+
   handleTitleOverrides = predefinedProps => ({
     onClick: (e, titleProps) => {
       e.preventDefault()
-      this.trySetState({
+      this.setState({
         open: !this.state.open,
       })
+      _.invoke(this.props, 'onOpenChanged', e, this.props)
       _.invoke(predefinedProps, 'onClick', e, titleProps)
     },
   })
 
   renderContent() {
     const { items, title, renderItemTitle } = this.props
-    const { open } = this.state
-
+    const open = this.props.exclusive ? this.props.open : this.state.open
```

Comment:
It looks like the open prop should be auto controlled prop, then you can make decision of when it should be changed. Even if it is auto controlled, you can control it from the parent component.